src/System.Private.CoreLib/shared/System/Threading/Tasks/ValueTask.cs

```diff
@@ -744,24 +744,28 @@ public bool IsCanceled
         public TResult Result
         {
             [MethodImpl(MethodImplOptions.AggressiveInlining)]
-            get
-            {
-                object? obj = _obj;
-                Debug.Assert(obj == null || obj is Task<TResult> || obj is IValueTaskSource<TResult>);
+            get => GetResult();
+        }
 
-                if (obj == null)
-                {
-                    return _result;
-                }
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        [StackTraceHidden]
+        internal TResult GetResult()
```

Comment:
The async generic machinery is contributing a ton to the footprint and startup. I do not think we want to be making this worse by creating helper methods like this.

Have you considered filtering out methods marked with AggressiveInlining in the pretty printer? It would actually allow you to strip StackTraceHidden from number of other places, and get more consistency in what one sees with optimizations on vs. off.

Reply:
I'd be happy with that

Reply:
https://github.com/dotnet/coreclr/pull/25408